time_manager/time_manager.F90

```diff
@@ -1213,9 +1213,27 @@ function real_to_time_type(x,err_msg) result(t)
   real :: tps
   real :: a
   tps = real(ticks_per_second)
-  a = x/spd
+
+  select type (x)
+  type is (real(kind=r4_kind))
+    a = x/spd
```

Comment:
are `a` and `spd` always going to be the same type as `x`?

Reply:
`a` and `spd` are real; their kind will depend on how FMS is compiled. Is there an issue if the kind of `a` and `spd` is not the same as the kind of `x`?

Reply:
This will produce a compiler warning if they types don't match
```
Warning: Possible change of value in conversion from REAL(8) to REAL(4) at (1) [-Wconversion]
```